src/controller.rs

```diff
@@ -41,8 +41,14 @@ impl<'b> Controller<'b> {
 
         let mut output_type = OutputType::from_mode(paging_mode, self.config.pager)?;
         let writer = output_type.handle()?;
-        let mut no_errors: bool = true;
 
+        let no_errors = self.run_with_writer(writer, handle_error);
+
+        Ok(no_errors)
+    }
+
+    pub fn run_with_writer(&self, mut writer: &mut dyn Write, handle_error: impl Fn(&Error)) -> bool {
```

Comment:
`writer` needs to reference a `&mut dyn Write`, but it doesn't have to be mutable itself, right? (which would allow retargeting the borrow).